source/slang/compiler.cpp

```diff
@@ -489,14 +464,11 @@ namespace Slang
         // The result for the translation unit will just be the concatenation
         // of the results for each entry point. This doesn't actually make
         // much sense, but it is good enough for now.
-        //
-        // TODO: Replace this with a packaged JSON and/or binary format.
```

Comment:
I don't think we can replace this `TODO` comment just yet. Just concatenating a bunch  of SPIRV or DXBC code doesn't really make for a usable container.

I'm fine with the "concatenate all the things" logic for now, with one big caveat: In the case of textual output, we need to make sure that the translation-unit result is only has one null terminator, at the end of all the output.

One way to handle all of these text-vs-binary differences is to have distinct `String outputSource` and `List<uint8_t> outputCode`. Looking at the edits, it might be worth considering.